lib/Service/QueueService.php

```diff
@@ -59,7 +60,12 @@ public function scheduleJob(QueueFile $file): void {
 	 * @throws \OCP\DB\Exception
 	 */
 	public function getFromQueue(int $storageId, int $rootId, int $batchSize): array {
-		return $this->queueMapper->getFromQueue($storageId, $rootId, $batchSize);
+		$nonUpdates = $this->queueMapper->getFromQueue($storageId, $rootId, $batchSize, true);
+		if (empty($nonUpdates)) {
+			return $this->queueMapper->getFromQueue($storageId, $rootId, $batchSize, false);
+		}
+
+		return $nonUpdates;
```

Comment:
sweet!

Reply:
can we add a stat entry for non-update files?
and maybe also judge initial indexing completion based on this?